net_buf: change return type of net_buf_max_len as size_t

MarkWangChinese opened the pull request:
Change net_buf_max_len to use size_t as return value type to align with the rest of the net_buf APIs.

MarkWangChinese committed:
net_buf: change return type of net_buf_max_len as size_t

Change net_buf_max_len to use size_t as return value type to align with
the rest of the net_buf APIs.

Signed-off-by: Mark Wang <yichang.wang@nxp.com>

henrikbrixandersen reviewed (commented):
This needs to go through the breaking API change process, right?

MarkWangChinese commented:
> @MarkWangChinese since you decided to open a new PR for this instead of keeping it in the other PR (and moving everything else out) it means that this will not fulfill the review time criteria before the feature freeze. Because of that it's essential to highlight that it's an API consistency fix. You're also missing the update to the migration guide as we discussed.

Thanks. I want to keep the https://github.com/zephyrproject-rtos/zephyr/pull/97440 to continue talking about large size support. Maybe creating another RFC to talk about it is better? Since the "breaking API" need to be clarified too, I think this change can't be merged before feature freeze.

kartben reviewed (changes requested):
It is a breaking change to a stable API (plus, it is not even mentioned in the migration guide). This doesn't sound right :)

MarkWangChinese committed:
doc: migration-guide-4.4: document net_buf API return type changes

Document the breaking API changes in net_buf_simple_max_len and
net_buf_max_len, which now return size_t instead of uint16_t to align
with other net_buf APIs.

Signed-off-by: Mark Wang <yichang.wang@nxp.com>

jhedberg commented:
The existence of this API has always been a bit awkward (the way it's defined feels arbitrary), so I think a good alternative option would be to deprecate and eventually remove it.

MarkWangChinese commented:
> The existence of this API has always been a bit awkward (the way it's defined feels arbitrary), so I think a good alternative option would be to deprecate and eventually remove it.

Thanks @jhedberg So, let me close this pr.
